wsclient/src/main/java/org/hpccsystems/ws/client/utils/Connection.java

```diff
@@ -1028,39 +1042,71 @@ public String sendHTTPRequest(String uri, String method) throws Exception
 
         URL url = new URL (getBaseUrl() + (uri != null && uri.startsWith("/") ? "" : "/") + uri);
 
-        HttpURLConnection httpURLConnection = (HttpURLConnection) url.openConnection(); //throws IOException
+        Span sendHTTPReqSpan = GlobalOpenTelemetry.get().getTracer(BaseHPCCWsClient.PROJECT_NAME)
+                .spanBuilder(method.toUpperCase() + " " + url.toExternalForm())
+                .setAttribute(ServerAttributes.SERVER_ADDRESS, getHost())
+                .setAttribute(ServerAttributes.SERVER_PORT, Long.getLong(getPort()))
+                .setAttribute(HttpAttributes.HTTP_REQUEST_METHOD, HttpAttributes.HttpRequestMethodValues.GET)
+                .setSpanKind(SpanKind.CLIENT)
+                .startSpan();
 
-         Connection.log.info("Sending HTTP " + method + "Request to:" + url.toString());
+        HttpURLConnection httpURLConnection = (HttpURLConnection) url.openConnection(); //throws IOException
 
-         if (hasCredentials())
-         {
-             httpURLConnection.setRequestProperty("Authorization", getBasicAuthString());
-         }
+        Connection.log.info("Sending HTTP " + method + "Request to:" + url.toString());
 
-         httpURLConnection.setRequestMethod(method); //throws ProtocolException
+        if (hasCredentials())
+        {
+            httpURLConnection.setRequestProperty("Authorization", getBasicAuthString());
+            sendHTTPReqSpan.setAttribute("hasCredentials", true);
+        }
+        else
+        {
+            sendHTTPReqSpan.setAttribute("hasCredentials", false);
+        }
 
-         int responseCode = httpURLConnection.getResponseCode(); //throws IOException
+        try (Scope scope = sendHTTPReqSpan.makeCurrent())
+        {
+            Span currentSpan = Span.current();
+            if (currentSpan != null && currentSpan.getSpanContext().isValid())
+            {
+                //overkill? we could just construct traceparent manually
+                Map<String, String> carrier = new HashMap<>();
+                TextMapSetter<Map<String, String>> setter = Map::put;
+                W3CTraceContextPropagator.getInstance().inject(Context.current(), carrier, setter);
+                String traceparent = carrier.getOrDefault("traceparent", "00-" + currentSpan.getSpanContext().getTraceId() + "-" + currentSpan.getSpanContext().getSpanId() + "-00");
```

Comment:
Would this make sense to move this functionality to create trace into a utility so that dfsclient can use this as well?